src/Workspaces/Core/Portable/CodeStyle/NotificationOption.cs

```diff
@@ -9,28 +9,23 @@
 using WorkspacesResources = Microsoft.CodeAnalysis.CodeStyleResources;
 #endif
 
-#if CODE_STYLE
-namespace Microsoft.CodeAnalysis.Internal.Options
-#else
 namespace Microsoft.CodeAnalysis.CodeStyle
-#endif
 {
-    /// <summary>
-    /// Offers different notification styles for enforcing
-    /// a code style. Under the hood, it simply maps to <see cref="DiagnosticSeverity"/>
-    /// </summary>
-    /// <remarks>
-    /// This also supports various properties for databinding.
-    /// </remarks>
-    /// <completionlist cref="NotificationOption"/>
+    /// <inheritdoc cref="NotificationOption2"/>
     public class NotificationOption
     {
-        public string Name { get; set; }
+        private readonly NotificationOption2 _notificationOptionImpl;
+
+        public string Name
+        {
+            get => _notificationOptionImpl.Name;
+            set => _notificationOptionImpl.Name = value;
```

Comment:
😨 Why do we allow setters here?

Reply:
Note these are shipped public APIs :-). I can file a tracking bug to triage these and mark the setters as obsolete.

Reply:
Filed https://github.com/dotnet/roslyn/issues/42550